tools/model_tools.R

```diff
@@ -0,0 +1,39 @@
+#' Interpolate missing (rodent) abundance data
+#' 
+#' @param abundance data table with new moon column
+#' @return data table of interpolation-inclusive counts for each species and 
+#'  total
+
+interpolate_abundance <- function(abundance){
+
+      moons <- (min(abundance$newmoonnumber)):(max(abundance$newmoonnumber))
+      nmoons <- length(moons)
+
+      species <- c("BA", "DM", "DO", "DS", "NA.", "OL", "OT", "PB", "PE", 
+                   "PF", "PH", "PL", "PM", "PP", "RF", "RM", "RO", "SF",
+                   "SH", "SO")
```

Comment:
Typing these in manually is fragile and confusing (I assumed this was an attempt to filter things for quite a while). I'm guessing this is actually just to get the species IDs, but in that case we're missing "PI".

It's better to extract this from the data itself:

```
species <- colnames(all)[2:(ncol(all)-4)]
```

This is still not ideal, which points towards an issue with the main data structure (we should use long, not wide, which will give us a column, named `species` to work with), but that's something to be fixed separately.

Reply:
totally with you on this one.

I grabbed that list from one of the species-level models scripts and just ported it into these ones. That's a bit troubling that we've got a missing species then?

Reply:
* PI is missing on purpose. It probably doesn't exist at the site, and should not have predictions made.
* Would it not be easier to just switch the data to long format now? Data-wise, it's mostly just a matter of adding `shape=flat` to the abundance function call. Then the models can be written that way from the start. It seems like a waste of time to merge them to work on wide data, then have to change them after the fact? Though, I know we don't want this PR to get too crazy. I could also submit a separate PR, before these go in?

Reply:
Oh, also, filtering out PI can be moved to the get data step in the process.

Reply:
oh! so do we not want PI include in the totals then, either?
and in that case, should PI not be getting pulled into the rodent data?

Reply:
and for the long vs wide...i'm inclined to hold off of that until we're through with what's being updated in these PRs and then I can double back and update the data format.

Reply:
where would I add the call to not include PI?

Reply:
In tools/model_functions.R `get_rodent_data`. (This is also where `shape=flat` would get added to `abundance()`.)

Right now, PI still gets included in the totals, because there actually were that many total target rodents in traps. It's just a species ID issue.

Reply:
solid thanks for pointing me there!

Reply:
as for including PI, with how the interpolation goes and due to Jensen's Inequality, the interpolation of the total is not equal to the total of the interpolations. That means, we should either [1] consider all of the species together and just interpolate the total for the four models (i.e. not fit species-specific models) or [2] restrict "total" to mean the total from the list of species included

Reply:
I think [2] is definitely fine. The interpretation of the `PI` ID (during the period of time it was used at all) should be "unknown Chaetodipus, ~1% chance it's actually a PI, probably a PP." And we already don't include other unknown Chaetodipus in the total, even though they count as target rodents too.

Reply:
ok great, thanks for your insight! 
I'll work on getting this implemented and let you know if I hit any more snags